codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-async-client-class.java

```diff
@@ -160,6 +160,7 @@ public final String serviceName() {
      * @see <a href="https://docs.aws.amazon.com/goto/WebAPI/json-service-2010-05-08/APostOperation" target="_top">AWS
      *      API Documentation</a>
      */
+    @Deprecated
```

Comment:
Looks like there's a `deprecatedMessage` field. Can we add it in the Javadocs?

Reply:
We can probably update this utility class to support operation models as well:

https://github.com/aws/aws-sdk-java-v2/blob/a8ec3f8c1aaf81e06394b30daec2074562581c70/codegen/src/main/java/software/amazon/awssdk/codegen/poet/model/DeprecationUtils.java#L34-L50

Reply:
Oh nice @Bennett-Lynch this looks perfect, will make some tweaks to use that.

@zoewangg sure thing! :)

Reply:
Added those changes. Let me know what you think!

Reply:
LGTM!